MediaBrowser.Providers/Music/MusicBrainzAlbumProvider.cs

```diff
@@ -714,34 +714,32 @@ private string GetFirstReleaseGroupId(XmlReader reader)
             return null;
         }
 
-        internal Task<HttpResponseInfo> GetMusicBrainzResponse(string url, bool isSearch, CancellationToken cancellationToken)
-        {
-            return GetMusicBrainzResponse(url, isSearch, false, cancellationToken);
-        }
-
         /// <summary>
-        /// Gets the music brainz response.
+        /// Makes request to MusicBrainz server and awaits a response.
         /// </summary>
-        internal async Task<HttpResponseInfo> GetMusicBrainzResponse(string url, bool isSearch, bool forceMusicBrainzProper, CancellationToken cancellationToken)
+        internal async Task<HttpResponseInfo> GetMusicBrainzResponse(string url, CancellationToken cancellationToken)
         {
-            var urlInfo = new MbzUrl(MusicBrainzBaseUrl, 1000);
-            var throttleMs = urlInfo.throttleMs;
+            // The Jellyfin user-agent is unrestricted but source IP must not exceed
+            // one request per second, therefore we rate limit to avoid throttling
+            // https://musicbrainz.org/doc/XML_Web_Service/Rate_Limiting
+            const long QueryIntervalMs = 1000u;
 
-            if (throttleMs > 0)
+            // Only delay if necessary
+            if (_stopWatchMusicBrainz.ElapsedMilliseconds < QueryIntervalMs)
             {
-                // MusicBrainz is extremely adamant about limiting to one request per second
-                _logger.LogDebug("Throttling MusicBrainz by {0}ms", throttleMs.ToString(CultureInfo.InvariantCulture));
-                await Task.Delay(throttleMs, cancellationToken).ConfigureAwait(false);
+                var delayMs = QueryIntervalMs - _stopWatchMusicBrainz.ElapsedMilliseconds;
+                await Task.Delay((int)delayMs, cancellationToken).ConfigureAwait(false);
             }
 
-            url = urlInfo.url.TrimEnd('/') + url;
+            _logger.LogDebug("MusicBrainz time since previous request: {0}ms", _stopWatchMusicBrainz.ElapsedMilliseconds);
```

Comment:
```suggestion
            _logger.LogDebug("MusicBrainz time since previous request: {0} ms", _stopWatchMusicBrainz.ElapsedMilliseconds);
```
also this logging is pointless in this place, as you've already waited for it to be more than your limit :) You should be logging before wait IMHO.

Reply:
This debug log entry is our evidence that we're meeting the MusicBrainz rate limit requirement..  I'm not interested if we delayed or not, just that we only call max once per sec.